Add release notes for 0.1.0

jashapiro opened the pull request:
As I was preparing #623, I thought it might be nice to have the release notes reviewed, which let me to wanting to create a changelog file to hold them all.

Please review these initial release notes, and let me know if there is anything you think I should specifically add or generally rephrase.

I also made a few updates fixing paths in templates that I noticed had changed and a few other minor template updates.

allyhawkins commented on CHANGELOG.md:
```diff
@@ -0,0 +1,35 @@
+
+With respect to infrastructure, the repository contains the following components:
+
+- detailed documentation in the `docs` directory explaining how to interact with the OpenScPCA project and how to set up and run analyses
```
Maybe provide a link to the actual docs too?

allyhawkins reviewed (approved):
I think this is fine, I just had one suggestion about including a link to the actual docs page, but up to you.

jaclyn-taroni commented on RELEASE_NOTES.md:
Why do we also have this file? What am I missing?

jashapiro replied:
I had named it that before. Not sure how it got duplicated rather than renamed!

jashapiro commented:
Okay, I removed the extra file, added the link @allyhawkins requested, and then decided that we should probably keep these root-level `.md` files spellchecked, so I added that pattern to the spellcheck action. Amazingly enough*, I didn't have any spelling errors, only module names that needed quoting. 

*I run `typos` locally, so not that amazing.

jaclyn-taroni reviewed (approved):
👍🏻
